storey/flow.py

```diff
@@ -988,26 +1005,28 @@ async def _save_key(self, table_path, key, aggr_item, additional_data=None):
         if not self._client_session:
             self._lazy_init()
 
-        request_data = self._get_attributes_as_blob(aggr_item)
-        if additional_data:
-            request_data.update(_v3io_build_put_item_request(additional_data))
+        request_data = self._generate_update_expression(aggr_item, additional_data)
 
-        data = {'Item': request_data, 'UpdateMode': 'CreateOrReplaceAttributes'}
+        data = {'UpdateExpression': request_data}
         response = await self._client_session.put(f'{self._webapi_url}/{table_path}/{key}',
-                                                  headers=self._put_item_headers, data=json.dumps(data), ssl=False)
+                                                  headers=self._update_item_headers, data=json.dumps(data), ssl=False)
         if not response.status == 200:
             body = await response.text()
             raise V3ioError(f'Failed to save aggregation for key: {key}. Response status code was {response.status}: {body}')
 
-    def _get_attributes_as_blob(self, aggregation_element):
-        data = {}
+    def _generate_update_expression(self, aggregation_element, additional_data):
+        expression = ''
+
         for name, bucket in aggregation_element.aggregation_buckets.items():
             # Only save raw aggregates, not virtual
             if bucket.should_persist:
                 blob = pickle.dumps(bucket.to_dict())
-                data[f'{self._aggregation_attribute_prefix}{name}'] = {'B': base64.b64encode(blob).decode('ascii')}
+                expression = expression + f"{self._aggregation_attribute_prefix}{name}=blob('{base64.b64encode(blob).decode('ascii')}');"
 
-        return data
+        if additional_data:
+            additional_expr = [f'{name}={_convert_python_obj_to_expression_value(value)}' for (name, value) in additional_data.items()]
+            expression = expression + ';'.join(additional_expr)
```

Comment:
Can unify the two concat methods so that you build one list and join it at the end.